### PR TITLE
mbed-os wifi netsocket tests with MBED_EXTENDED_TESTS macro

### PR DESCRIPTION
**Pull request by jeromecoutant**
MBED_EXTENDED_TESTS macro needs few update in our wifi driver.

Test result is OK now!

+-------------------------+---------------------+---------------------+------------------------------------+--------+--------+--------+--------------------+
| target                  | platform_name       | test suite          | test case                          | passed | failed | result | elapsed_time (sec) |
+-------------------------+---------------------+---------------------+------------------------------------+--------+--------+--------+--------------------+
| DISCO_L475VG_IOT01A-ARM | DISCO_L475VG_IOT01A | tests-netsocket-tcp | TCPSOCKET_CONNECT_INVALID          | 1      | 0      | OK     | 0.12               |
| DISCO_L475VG_IOT01A-ARM | DISCO_L475VG_IOT01A | tests-netsocket-tcp | TCPSOCKET_ECHOTEST                 | 1      | 0      | OK     | 2.8                |
| DISCO_L475VG_IOT01A-ARM | DISCO_L475VG_IOT01A | tests-netsocket-tcp | TCPSOCKET_ECHOTEST_BURST           | 1      | 0      | OK     | 13.5               |
| DISCO_L475VG_IOT01A-ARM | DISCO_L475VG_IOT01A | tests-netsocket-tcp | TCPSOCKET_ECHOTEST_BURST_NONBLOCK  | 1      | 0      | OK     | 16.53              |
| DISCO_L475VG_IOT01A-ARM | DISCO_L475VG_IOT01A | tests-netsocket-tcp | TCPSOCKET_ECHOTEST_NONBLOCK        | 1      | 0      | OK     | 23.03              |
| DISCO_L475VG_IOT01A-ARM | DISCO_L475VG_IOT01A | tests-netsocket-tcp | TCPSOCKET_ENDPOINT_CLOSE           | 0      | 1      | OK     | 0.68               |
| DISCO_L475VG_IOT01A-ARM | DISCO_L475VG_IOT01A | tests-netsocket-tcp | TCPSOCKET_OPEN_CLOSE_REPEAT        | 1      | 0      | OK     | 0.06               |
| DISCO_L475VG_IOT01A-ARM | DISCO_L475VG_IOT01A | tests-netsocket-tcp | TCPSOCKET_OPEN_LIMIT               | 1      | 0      | OK     | 0.23               |
| DISCO_L475VG_IOT01A-ARM | DISCO_L475VG_IOT01A | tests-netsocket-tcp | TCPSOCKET_RECV_100K                | 1      | 0      | OK     | 2.45               |
| DISCO_L475VG_IOT01A-ARM | DISCO_L475VG_IOT01A | tests-netsocket-tcp | TCPSOCKET_RECV_100K_NONBLOCK       | 1      | 0      | OK     | 2.78               |
| DISCO_L475VG_IOT01A-ARM | DISCO_L475VG_IOT01A | tests-netsocket-tcp | TCPSOCKET_RECV_TIMEOUT             | 1      | 0      | OK     | 1.54               |
| DISCO_L475VG_IOT01A-ARM | DISCO_L475VG_IOT01A | tests-netsocket-tcp | TCPSOCKET_SEND_REPEAT              | 1      | 0      | OK     | 5.1                |
| DISCO_L475VG_IOT01A-ARM | DISCO_L475VG_IOT01A | tests-netsocket-tcp | TCPSOCKET_SEND_TIMEOUT             | 1      | 0      | OK     | 0.12               |
| DISCO_L475VG_IOT01A-ARM | DISCO_L475VG_IOT01A | tests-netsocket-tcp | TCPSOCKET_THREAD_PER_SOCKET_SAFETY | 1      | 0      | OK     | 1.24               |
+-------------------------+---------------------+---------------------+------------------------------------+--------+--------+--------+--------------------+

+------------------+---------------+---------------------+------------------------------------+--------+--------+--------+--------------------+
| target           | platform_name | test suite          | test case                          | passed | failed | result | elapsed_time (sec) |
+------------------+---------------+---------------------+------------------------------------+--------+--------+--------+--------------------+
| DISCO_F413ZH-ARM | DISCO_F413ZH  | tests-netsocket-tcp | TCPSOCKET_CONNECT_INVALID          | 1      | 0      | OK     | 0.32               |
| DISCO_F413ZH-ARM | DISCO_F413ZH  | tests-netsocket-tcp | TCPSOCKET_ECHOTEST                 | 1      | 0      | OK     | 1.93               |
| DISCO_F413ZH-ARM | DISCO_F413ZH  | tests-netsocket-tcp | TCPSOCKET_ECHOTEST_BURST           | 1      | 0      | OK     | 14.85              |
| DISCO_F413ZH-ARM | DISCO_F413ZH  | tests-netsocket-tcp | TCPSOCKET_ECHOTEST_BURST_NONBLOCK  | 1      | 0      | OK     | 12.09              |
| DISCO_F413ZH-ARM | DISCO_F413ZH  | tests-netsocket-tcp | TCPSOCKET_ECHOTEST_NONBLOCK        | 1      | 0      | OK     | 23.25              |
| DISCO_F413ZH-ARM | DISCO_F413ZH  | tests-netsocket-tcp | TCPSOCKET_ENDPOINT_CLOSE           | 1      | 0      | OK     | 0.63               |
| DISCO_F413ZH-ARM | DISCO_F413ZH  | tests-netsocket-tcp | TCPSOCKET_OPEN_CLOSE_REPEAT        | 1      | 0      | OK     | 0.07               |
| DISCO_F413ZH-ARM | DISCO_F413ZH  | tests-netsocket-tcp | TCPSOCKET_OPEN_LIMIT               | 1      | 0      | OK     | 0.3                |
| DISCO_F413ZH-ARM | DISCO_F413ZH  | tests-netsocket-tcp | TCPSOCKET_RECV_100K                | 1      | 0      | OK     | 3.33               |
| DISCO_F413ZH-ARM | DISCO_F413ZH  | tests-netsocket-tcp | TCPSOCKET_RECV_100K_NONBLOCK       | 1      | 0      | OK     | 3.36               |
| DISCO_F413ZH-ARM | DISCO_F413ZH  | tests-netsocket-tcp | TCPSOCKET_RECV_TIMEOUT             | 1      | 0      | OK     | 1.17               |
| DISCO_F413ZH-ARM | DISCO_F413ZH  | tests-netsocket-tcp | TCPSOCKET_SEND_REPEAT              | 1      | 0      | OK     | 5.85               |
| DISCO_F413ZH-ARM | DISCO_F413ZH  | tests-netsocket-tcp | TCPSOCKET_SEND_TIMEOUT             | 1      | 0      | OK     | 0.24               |
| DISCO_F413ZH-ARM | DISCO_F413ZH  | tests-netsocket-tcp | TCPSOCKET_THREAD_PER_SOCKET_SAFETY | 1      | 0      | OK     | 1.37               |
+------------------+---------------+---------------------+------------------------------------+--------+--------+--------+--------------------+

### CODE REVIEW

File: ISM43362Interface.cpp [hunk 1]
```diff
@@ -424,7 +424,7 @@ int ISM43362Interface::socket_send_nolock(void *handle, const void *data, unsign
 
     if (!_ism.send(socket->id, data, size)) {
         debug_if(_ism_debug, "ISM43362Interface: socket_send ERROR\r\n");
```
Comment (LMESTM) [inline ISM43362Interface.cpp]: do we have enough information from WIFI module to distinguish btw WOULD_BLOCK and ERROR

File: ISM43362Interface.cpp [hunk 2]
```diff
@@ -441,7 +441,7 @@ int ISM43362Interface::socket_recv(void *handle, void *data, unsigned size)
 
     if (!socket->connected) {
         _mutex.unlock();
-        return NSAPI_ERROR_CONNECTION_LOST;
+        return 0;
     }
```
Comment (LMESTM) [inline ISM43362Interface.cpp]: question: socket_recv is supposed to return
     *  @return         Number of received bytes on success, negative error
     *                  code on failure
here, if the socket is not valid, shouldn't it be considered a failure ?

File: ISM43362Interface.cpp [hunk 3]
```diff
@@ -454,7 +454,7 @@ int ISM43362Interface::socket_recv(void *handle, void *data, unsigned size)
         } else if (read_amount < 0) {
             socket->connected = false;
             _mutex.unlock();
-            return NSAPI_ERROR_CONNECTION_LOST;
+            return 0;
         }
```
Comment (LMESTM) [inline ISM43362Interface.cpp]: Same here. Question: socket_recv is supposed to return
     *  @return         Number of received bytes on success, negative error
     *                  code on failure
here, if the socket is not valid, shouldn't it be considered a failure ?

File: ISM43362/ATParser/BufferedSpi/BufferedSpi.cpp [hunk 4]
```diff
@@ -257,16 +258,18 @@ ssize_t BufferedSpi::read(uint32_t max)
     }
 
     enable_nss();
-    while (dataready.read() == 1 && (len < (_buf_size - 1))) {
+    while (dataready.read() == 1 && (len < (_buf_size - 2))) {
         tmp = SPI::write(0xAA);  // dummy write to receive 2 bytes
 
-        if (!((len == 0) && (tmp == 0x0A0D))) {
+        if (!((len == 0) && (tmp == 0x0A0D) && (FirstRemoved))) {
```
Comment (LMESTM) [inline ISM43362/ATParser/BufferedSpi/BufferedSpi.cpp]: Ok - could be useful to explain that DATA can be indeed starting with 0x0A0D  ... well done !!!

File: ISM43362Interface.cpp (hunk 1 above)
Comment (jeromecoutant) [reply ISM43362Interface.cpp]: It seems that this patch is, in fact, not necessary to make TCP test passed!
Let's keep the code with the error code.

**Comment by jeromecoutant** [comment]
From the 5 previous commits, I removed 
- the one which is not needed (WOULD_BLOCK in the send procedure)
- the one which need more analysis (NSAPI_ERROR_CONNECTION_LOST sub case). It will be part of a specific PR later.

**Pull request merged**